File: typescript/src/repl.ts
import { stdin as input, stdout as output } from 'node:process';
import * as readline from 'node:readline/promises';
import { Lexer } from "./lexer";
import { Token, TokenTypes } from "./token";

async function start(){
    const rl = readline.createInterface({ input, output, prompt: ">> " });
    rl.prompt();
    rl.on('line', (line) => {
        const lexer = new Lexer(line);
        let token: Token | undefined;

        while(token?.type !== TokenTypes.EOF) {
            token = lexer.nextToken();
            console.log(token);
        }

        rl.prompt();
    });
}

start();

File: rust/src/repl/repl.rs
use crate::lexer::lexer::Lexer;
use crate::lexer::token::Token;
use std::io::stdin;

pub fn start() {
    stdin().lines().for_each(|line| {
        if let Ok(line) = line {
            let mut lexer = Lexer::new(line);

            loop {
                let tok = lexer.next_token();
                if tok == Token::Eof {
                    break;
                }

                println!("{} ", tok);
                //println!("Token: {},", tok);
            }
        }
    });
}
